client/src/containers/CreateProjectModal/index.tsx

```diff
@@ -54,14 +55,35 @@ const CreateProjectModal: React.FC = () => {
 		);
 	};
 
+	const onCreateProject = (): void => {
+		if (!isKeyTouched) {
+			const isGeneratedKeyValid = validator.isLength(key, { min: 2 });
+			if (isGeneratedKeyValid && isNameValid) {
+				startCreatingProject();
+			}
+		}
+
+		if (!isNameValid || !isKeyValid) {
+			setIsValidErrorShown(true);
+			return;
+		}
+		startCreatingProject();
+	};
+
 	const generateKey = (name: string): string => {
 		let result = key;
 		if (!isKeyTouched) {
-			result = name
-				.split(' ')
-				.filter(Boolean)
-				.map((word) => word[0].toUpperCase())
-				.join('');
+			const isSpace = name.trimEnd().search(' ');
```

Comment:
Can _isSpace_ be renamed to _spaceIndex_?

Reply:
Renamed